disnake/user.py

```diff
@@ -142,8 +152,17 @@ def avatar(self) -> Optional[Asset]:
 
     @property
     def default_avatar(self) -> Asset:
-        """:class:`Asset`: Returns the default avatar for a given user. This is calculated by the user's discriminator."""
-        return Asset._from_default_avatar(self._state, int(self.discriminator) % len(DefaultAvatar))
+        """:class:`Asset`: Returns the default avatar for a given user.
+
+        .. versionchanged:: 2.9
+            Added handling for users migrated to the new username system without discriminators.
+        """
+        if self.discriminator == "0":
+            num = self.id >> 22
+        else:
+            # legacy behavior
+            num = int(self.discriminator)
+        return Asset._from_default_avatar(self._state, num % len(DefaultAvatar))
```

Comment:
have we confirmed this is accurate, and they aren't taking this opportunity to add avatars/5.png?

Reply:
The documentation uses `% 5`, the client also references five default assets, and I haven't heard anything about this being changed, so no